### PR TITLE
Add unit tests for subtraction, square root and xml parser

### PR DESCRIPTION
**Pull request by CipDan**

### CODE REVIEW

File: test/models/square-root.test.ts [hunk 1]
```diff
@@ -27,5 +27,19 @@ describe("SquareRoot.apply", () => {
...
+    });
+
+    test("sqrt(1048576) = 1024", () => {
+        const testNumber: number[] = [7, 6, 7, 5, 8, 4, 0, 1];
+        const result: number[] = [4, 4, 2, 0, 1];
+
+        expect(new SquareRoot().apply(testNumber)).toEqual(result);
+    });
+
+    test("sqrt(1234567) approximately equal to 1111", () => {
```
Comment (IoanSava) [inline test/models/square-root.test.ts:39]: ```sqrt(1234567) = 1111``` is enough

File: test/models/subtraction.test.ts [hunk 2]
```diff
@@ -9,6 +9,8 @@ describe("Subtraction.apply", () => {
         expect(() => new Subtraction().apply(minuend, subtrahend)).toThrow(new NegativeResultOfSubtractionError(minuend, subtrahend));
     });
 
+    // Test is obsolete here, but checks lines 27, 31 for big-number.ts
```
Comment (IoanSava) [inline test/models/subtraction.test.ts:12]: You can remove the comments.

File: test/services/data/xml-parser/example1.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+		</variable>
+		<variable>
+			<name>b</name>
+			<value>2</value>
+		</variable>
+		<variable>
+			<name>c</name>
+			<value>30</value>
+		</variable>
+        <variable>
```
Comment (IoanSava) [inline test/services/data/xml-parser/example1.xml:17]: Fix indentation for this line

Comment (CipDan) [reply test/services/data/xml-parser/example1.xml:17]: Which one?

File: test/services/xml-parser.test.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { XMLParser } from "../../src/services/xml-parser";
+import { join } from "path";
+import * as fs from "fs";
+
+describe("XMLParser", () => {
```
Comment (IoanSava) [inline test/services/xml-parser.test.ts:9]: You should have two ```describe``` blocks inside this one:

``` javascript
describe("XMLParser", ...

describe("extractExpression", 
// extractExpression related tests
);

describe("extractVariables",
// extractVariables related tests
);

);
```

**Comment by IoanSava** [comment]
Ship it
```
                                  )___(
                           _______/__/_
                  ___     /===========|   ___
 ____       __   [\\\]___/____________|__[///]   __
 \   \_____[\\]__/___________________________\__[//]___
  \40A                                                 |
   \                                                  /
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by IoanSava — APPROVED** [review]

**Review by DianaCrainic — APPROVED** [review]

**Pull request merged**